[TECH] Faciliter le déploiement sur Scalingo (#6519)
1024pix/pix: DEPLOY.md

# Scalingo

> These directives are intended to be used with NodeJs buildpack

Reviewer: ```suggestion
These directives are intended to be used with a NodeJs buildpack.
```

Author: Corrigé

## Create applications

### Add-ons

> Add Postgresql & redis addons

Reviewer: ```suggestion
Add a Postgresql and a redis addon.
```

Author: Corrigé

> Add Postgresql & redis addons

Reviewer 2: Sur l'API uniquement

Author: Corrigé

### API

> Specify the application to deploy: `PROJECT_DIR=api`

Reviewer: ```suggestion
Specify the application folder used for deployment: `PROJECT_DIR=api`
```

Author: Corrigé

> Deploy a first version to initialize the database

Reviewer: ```suggestion
The first API deployment will initialize the database.
```

Author: Corrigé

### Front-ends

Enable build:  `NPM_CONFIG_PRODUCTION=false`

> Specify the application to deploy: `PROJECT_DIR=mon-pix`

Reviewer: ```suggestion
Specify the application folder used for deployment: `PROJECT_DIR=mon-pix`
```

Author: Corrigé

Build production-like asset:  `BUILD_ENVIRONMENT=production`
Setup API proxy: `API_HOST=https://<APPLICATION-NAME>.<REGION>.scalingo.io`

### Check connectivity

Create an account on pix-app.

## Bootstrap

### Get administrator privileges

> To be able to create campaign, you'll need to create organization.

Reviewer: ```suggestion
To be able to create campaigns, you'll need to create an organization.
```

Author: Corrigé

These can only be created in pix-admin.

> To get access to pix-admin, grant the relevant privileges to an existing user through SQL

Reviewer: ```suggestion
To get access to pix-admin, grant the relevant privileges to an existing user through SQL :
```

Author: Corrigé

```shell

> scalingo --region <REGION> --app <APPLICATION-NAME> pgsql-console

Reviewer 2: Il faut télécharger le CLI scalingo d'abord

Author: Corrigé

```

``` postgres-sql
SELECT u.id FROM users u WHERE u.email = '<EMAIL>';

INSERT INTO
"pix-admin-roles" ("userId", "role")
VALUES
(<USER_ID>, 'SUPER_ADMIN');
```

### Plug LCMS preview

> Get an admin account
> 
> Add these environment variables in LCMS

Reviewer: ```suggestion
Using an admin account previously created on Pix as values, add these environment variables in LCMS :
```

Author: Corrigé

```dotenv
# No trailing slash !
PIX_API_BASEURL=https://<APPLICATION-NAME>.<REGION>.scalingo.io
PIX_API_USER_EMAIL=<ACCOUNT-EMAIL>
PIX_API_USER_PASSWORD=<ACCOUNT-PASSWORD>
```

# Github

## Auto-merge

> Add in repository secret

Reviewer: ```suggestion
Add the following variable with a valid GitHub token for a user with repository write access in the repository action secrets :
```

Author: Corrigé

```
PIX_SERVICE_ACTIONS_TOKEN
```

## Configuration file change

> Add in repository secrets

Reviewer: ```suggestion
Add the following variable with a valid Slack token and the scalingo link to the application in the repository action secrets :
```

Author: Corrigé
